lib/emoji.rb

```diff
@@ -63,19 +63,24 @@ def edit_emoji(emoji)
 
   # Public: Find an emoji by its aliased name. Return nil if missing.
   def find_by_alias(name)
-    names_index[name]
+    @mutex.synchronize do
```

Comment:
@mislav check  `all unless defined @all` is in each methods `names_index`/`unicode_index`, and if we add mutex to only `all`method, we will not avoid problem, because `@all` will be defined, but `@names_index` and `@unicodes_index` haven't yet, and second thread catch the NoMethodError.